jerry-core/ecma/builtin-objects/ecma-builtin-regexp-prototype.c

```diff
@@ -744,7 +744,7 @@ ecma_builtin_regexp_prototype_dispatch_routine (uint8_t builtin_routine_id, /**<
     {
       if (!ecma_object_class_is (obj_p, LIT_MAGIC_STRING_REGEXP_UL))
       {
-        if (ecma_builtin_is (obj_p, ECMA_BUILTIN_ID_REGEXP_PROTOTYPE))
```

Comment:
`ecma_builtin_is ` is a widely used function, I think we should update it to properly support built-in objects from different realms. Maybe it would fix other testcases as well.

Reply:
`builtin_get` checks type regardless of realm. Is this the problem? A regexp coming from a different realm is not a regexp?

Reply:
@zherczeg The key from the standard is `If SameValue(R, %RegExpPrototype%)`

Where `%RegExpPrototype%` comes from the current realm's intrinsics.  

```js
/*---
esid: sec-get-regexp.prototype.dotall
description: Invoked on a cross-realm object
info: |
    get RegExp.prototype.dotAll

    1. Let R be the this value.
    2. If Type(R) is not Object, throw a TypeError exception.
    3. If R does not have an [[OriginalFlags]] internal slot, then
      a. If SameValue(R, %RegExpPrototype%) is true, return undefined.
      b. Otherwise, throw a TypeError exception.
features: [regexp-dotall, cross-realm]
---*/

var dotAll = Object.getOwnPropertyDescriptor(RegExp.prototype, 'dotAll').get;
var other = $262.createRealm().global;
var otherRegExpProto = other.RegExp.prototype;
var otherRegExpGetter = Object.getOwnPropertyDescriptor(otherRegExpProto, 'dotAll').get;

assert.throws(TypeError, function() {
  dotAll.call(otherRegExpProto);
}, 'cross-realm RegExp.prototype');

assert.throws(other.TypeError, function() {
  otherRegExpGetter.call(RegExp.prototype);
}, 'cross-realm RegExp.prototype getter method against primary realm RegExp.prototype');
```

Reply:
In short, a regex from another realm is regex, but the regex prototype from another realm is not a regexp prototype?